### ISSUE TITLE
Catching exists gracefully (question/discussion)

### ISSUE THREAD

**Issue by bogdan-dumitru** [antonmi/espec#171]
I'm super new to elixir and the whole ecosystem but I'm getting a really nice and tingly feeling about it all. I'm mostly curious about something small I've noticed and am asking this (selfishly) mainly to further my own understanding.

### Context

 I'm in the middle of the elixir/otp tutorial, the KV application, and half-way through I decided to switch from ExUnit to ESpec converting the specs I had already written and move forward writing them in ESpec (coming from ruby it made sense that I would end up using this anyway). It's funny that the spec I hadn't fixed yet was one to do with directly linked child agents causing a Registry agent to fail.

The output of running the ``ExUnit`` tests:
```
....

  1) test removes bucket on crash (KV.RegistryTest)
     test/kv/registry_test.exs:26
     ** (EXIT from #PID<0.157.0>) shutdown

.

Finished in 0.09 seconds (0.09s on load, 0.00s on tests)
6 tests, 1 failure

Randomized with seed 296261
```

The output of running the converted ``ESpec`` tests:
```
...** (EXIT from #PID<0.47.0>) shutdown
```

### Question...

As I said I'm just wrapping my head around everything but my instinct is telling me that that things get linked at the test framework level in such a way that it doesn't gracefully handle specs failing with exists and the whole supervisor goes down (feel free to correct any and all semantic mishaps) causing basically the test suite to never actually finish and report full results.

Basically I'm wondering:

- Is this a design choice?
- Is it something that I can fix by configuring my test suite differently?
- Is it something that needs to be addressed at the framework level?
- What is Sponge Bob most scared of?

**Comment by antonmi** [comment]
Hi, @bogdan-dumitru !
As I understand, your spec crashes ESpec process.
May you please provide an example of the code you are testing?
It is very difficult to say is a problem in ESpec or in your test.

**Comment by bogdan-dumitru** [comment]
Gotcha, Yup I'll provide a quick repo with Essex and exunit tests that
highlight what I'm talking about. I'm currently afk but will get that up
tomorrow. Thanks!
